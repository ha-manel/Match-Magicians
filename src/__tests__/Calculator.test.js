import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Calculator from '../components/calculator';
import '@testing-library/jest-dom';

describe('Testing the calculator', () => {
  it('Calculator is rendered', () => {
    const calc = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calc).toMatchSnapshot();
  });

  it('Test if doing 10+22=32', () => {
    const { container } = render(<Calculator />);

    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));

    const result = container.querySelector('.calc-result').innerHTML;
    expect(result).toEqual('32');
  });
});
