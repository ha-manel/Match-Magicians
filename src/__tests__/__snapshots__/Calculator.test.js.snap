// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the calculator Calculator is rendered 1`] = `
<div
  id="calc-section"
>
  <h2
    id="calculator-header"
  >
    Let's do some math!
  </h2>
  <div
    className="calc-container"
  >
    <div
      className="calc-result"
    >
        
    </div>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calc-operators"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calc-operators"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calc-operators"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calc-operators"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calc-operators"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
